src/Governance/MultichainGovernor/MultichainGovernor.sol

```diff
@@ -0,0 +1,881 @@
+pragma solidity 0.8.19;
+
+import {GovernorCompatibilityBravoUpgradeable} from "@openzeppelin-contracts-upgradeable/contracts/governance/compatibility/GovernorCompatibilityBravoUpgradeable.sol";
+import {GovernorSettingsUpgradeable} from "@openzeppelin-contracts-upgradeable/contracts/governance/extensions/GovernorSettingsUpgradeable.sol";
+import {ERC20VotesCompUpgradeable} from "@openzeppelin-contracts-upgradeable/contracts/token/ERC20/extensions/ERC20VotesCompUpgradeable.sol";
+import {GovernorUpgradeable} from "@openzeppelin-contracts-upgradeable/contracts/governance/GovernorUpgradeable.sol";
+import {EnumerableSet} from "@openzeppelin-contracts/contracts/utils/structs/EnumerableSet.sol";
+
+import {xWELL} from "@protocol/xWELL/xWELL.sol";
+import {IWormholeRelayer} from "@protocol/wormhole/IWormholeRelayer.sol";
+import {SnapshotInterface} from "@protocol/Governance/MultichainGovernor/SnapshotInterface.sol";
+import {IMultichainGovernor} from "@protocol/Governance/MultichainGovernor/IMultichainGovernor.sol";
+import {WormholeTrustedSender} from "@protocol/Governance/WormholeTrustedSender.sol";
+import {ConfigurablePauseGuardian} from "@protocol/xWELL/ConfigurablePauseGuardian.sol";
+
+/// WARNING: this contract is at very high risk of running over bytecode size limit
+///   we may need to split things out into multiple contracts, so keep things as
+///   concise as possible.
+
+/// @notice pauseable by the guardian
+/// @notice upgradeable, constructor disables implementation
+
+/// Note:
+/// - moonbeam block times are consistently 12 seconds with few exceptions https://moonscan.io/chart/blocktime
+/// this means that a timestamp can be converted to a block number with a high degree of accuracy
+
+/// there can only be one
+/// TODO, remove all OZ governor stuff, we're going to roll our own from scratch
+abstract contract MultichainGovernor is
+    WormholeTrustedSender,
+    IMultichainGovernor,
+    ConfigurablePauseGuardian
+{
+    using EnumerableSet for EnumerableSet.UintSet;
+
+    /// --------------------------------------------------------- ///
+    /// --------------------------------------------------------- ///
+    /// ----------------------- CONSTANTS ----------------------- ///
+    /// --------------------------------------------------------- ///
+    /// --------------------------------------------------------- ///
+
+    /// @notice Values for votes
+
+    /// @notice value for a yes vote
+    uint8 public constant VOTE_VALUE_YES = 0;
+
+    /// @notice value for a no vote
+    uint8 public constant VOTE_VALUE_NO = 1;
+
+    /// @notice value for an abstain vote
+    uint8 public constant VOTE_VALUE_ABSTAIN = 2;
```

Comment:
@ElliotFriedman shall we move these constants to another file so we can reuse it on MultichainVoteCollection?